commands/scan/scan.go

```diff
@@ -403,49 +385,58 @@ func (scanCmd *ScanCommand) createIndexerHandlerFunc(file *spec.File, entitledFo
-					jfrogAppsConfig, err := jas.CreateJFrogAppsConfig(workingDirs)
-					if err != nil {
-						log.Error(fmt.Sprintf("failed to create JFrogAppsConfig: %s", err.Error()))
-						indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
-					}
-					scanner := &jas.JasScanner{}
-					scanner, err = jas.CreateJasScanner(scanner, jfrogAppsConfig, scanCmd.serverDetails, jas.GetAnalyzerManagerXscEnvVars("", techutils.Technology(graphScanResults.ScannedPackageType)))
-					if err != nil {
-						log.Error(fmt.Sprintf("failed to create jas scanner: %s", err.Error()))
-						indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
-					}
-					err = runner.AddJasScannersTasks(jasFileProducerConsumer, &scanResults, &depsList, scanCmd.serverDetails, false, scanner, applicability.ApplicabilityDockerScanScanType, secrets.SecretsScannerDockerScanType, jasErrHandlerFunc, utils.GetAllSupportedScans())
-					if err != nil {
-						log.Error(fmt.Sprintf("scanning '%s' failed with error: %s", graph.Id, err.Error()))
-						indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
-					}
+				// Run Jas scans
+				scanner, err := getJasScanner(filePath, scanCmd.serverDetails, targetResults)
+				if err != nil {
+					return err
+				}
+				module, err := getJasModule(targetResults)
+				if err != nil {
+					return err
+				}
+				err = runner.AddJasScannersTasks(jasFileProducerConsumer, module, targetResults, directDepsListFromVulnerabilities(*graphScanResults), scanCmd.serverDetails, false, scanner, applicability.ApplicabilityDockerScanScanType, secrets.SecretsScannerDockerScanType, utils.GetAllSupportedScans())
+				if err != nil {
+					log.Error(fmt.Sprintf("%s jas scanning failed with error: %s", clientutils.GetLogMsgPrefix(scanThreadId, false), err.Error()))
+					targetResults.AddError(err)
 				}
-				resultsArr[threadId] = append(resultsArr[threadId], &ScanInfo{Target: filePath, Result: graphScanResults, ExtendedScanResults: scanResults.ExtendedScanResults})
 				return
 			}
-
 			_, _ = indexedFileProducer.AddTask(taskFunc)
 			return
 		}
 	}
 }
 
+func getJasScanner(_ string, serverDetails *config.ServerDetails, targetResults *results.TargetResults) (*jas.JasScanner, error) {
+	scanner, err := jas.CreateJasScanner(&jas.JasScanner{}, serverDetails, jas.GetAnalyzerManagerXscEnvVars("", targetResults.GetTechnologies()...))
+	if err != nil {
+		log.Error(fmt.Sprintf("failed to create jas scanner: %s", err.Error()))
+		targetResults.AddError(err)
+		return nil, err
+	}
+	return scanner, nil
+}
+
+func getJasModule(targetResults *results.TargetResults) (jfrogappsconfig.Module, error) {
+	jfrogAppsConfig, err := jas.CreateJFrogAppsConfig([]string{targetResults.Target})
```

Comment:
why are we keeping the logic of using JfrogAppsConfig in the refactor? it is going to be deprecated soon
We should decrease the usage of it anywhere we can and soon to get rid of this concept completely. dont we?

Reply:
JfrogAppsConfig will stay for now, but dealing with that will be in the next PR